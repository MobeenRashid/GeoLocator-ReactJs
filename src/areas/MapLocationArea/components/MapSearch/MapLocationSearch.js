import React, { Component } from 'react';
import { Input } from 'antd';
import GoogleMapSearch from '../../../../components/GoogleMap/GoogleMapSearch';
const defaultState = {
    defaultLocation: null,
    searchInput: null
}

class MapLocationSearch extends Component {

    constructor(props) {
        super(props);
        this.state = defaultState;

        this.onLocationSelect = this.onLocationSelect.bind(this);
        this.onChange = this.onChange.bind(this);
    }

    static getDerivedStateFromProps(props, state) {
        if (props.reset) {
            return defaultState;
        }

        let searchInput;
        if (state.searchInput === null || state.searchInput === undefined) {
            searchInput = (props.defaultLocation && props.defaultLocation.address);
        }
        return {
            defaultLocation: props.defaultLocation,
            searchInput: searchInput || state.searchInput
        }
    }

    componentDidMount() {
        let { defaultLocation } = this.state;
        defaultLocation = defaultLocation || this.props.defaultLocation;
        let searchInput = (defaultLocation && defaultLocation.address);

        this.setState({ searchInput });
    }


    onChange(e) {
        if (e && e.target) {
            this.setState({ searchInput: e.target.value });
        }
    }

    onLocationSelect(place) {
        if (place) {
            let selectedLocation = {
                id: place.place_id,
                address: place.formatted_address,
                lat: place.geometry.location.lat(),
                lng: place.geometry.location.lng()
            };
            this.props.onLocationSelect(selectedLocation);
            this.setState({ searchInput: selectedLocation.address });
        }
    }

    render() {
        let { searchInput } = this.state;
        return (
            <div className="search-box-holder">
                <GoogleMapSearch
                    onPlaceSelect={this.onLocationSelect}>
                    <Input className="location-input" disabled={this.props.disabled} placeholder="search location" value={searchInput} onChange={this.onChange}>
                    </Input>
                </GoogleMapSearch>
            </div>);
    }
}
export default MapLocationSearch;