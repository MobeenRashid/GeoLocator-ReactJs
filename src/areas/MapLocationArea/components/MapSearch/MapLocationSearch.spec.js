import React from 'react';
import { mount } from 'enzyme';
import TestRenderer from 'react-test-renderer';
import MapLocationSearch from './MapLocationSearch';

describe("Map location search", () => {

    test("Should render correctly", () => {
        const component = mount(<MapLocationSearch></MapLocationSearch>);
        expect(component).toMatchSnapshot();
        component.unmount();
    });

    test.skip('Should accept user input correctly', () => {
        const component = TestRenderer.create(<MapLocationSearch></MapLocationSearch>);
        console.log(component.toJSON());

        const inputSearch = component.root.findByProps({ className: 'location-input' });
        inputSearch.props.onChange({ target: { value: 'Berline' } });
        expect(inputSearch.props.value).toBe('Berline');
    });
});