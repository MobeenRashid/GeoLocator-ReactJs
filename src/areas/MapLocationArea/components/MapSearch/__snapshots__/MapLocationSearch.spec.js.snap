// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map location search Should render correctly 1`] = `
<MapLocationSearch>
  <div
    className="search-box-holder"
  >
    <GoogleMapSearch
      onPlaceSelect={[Function]}
    >
      <withScriptjs(withGoogleMap(Component))
        containerElement={
          <div
            style={
              Object {
                "height": "0",
              }
            }
          />
        }
        googleMapURL="https://maps.googleapis.com/maps/api/js?key=AIzaSyCDqC2TJWsyldn46K-rzxisTo9YRACImBE&v=3.exp&libraries=geometry,drawing,places"
        loadingElement={
          <div
            style={
              Object {
                "height": "100%",
              }
            }
          />
        }
        mapElement={
          <div
            style={
              Object {
                "height": "100%",
              }
            }
          />
        }
        onPlacesChanged={[Function]}
        onSearchBoxMounted={[Function]}
      >
        <div
          style={
            Object {
              "height": "100%",
            }
          }
        />
      </withScriptjs(withGoogleMap(Component))>
    </GoogleMapSearch>
  </div>
</MapLocationSearch>
`;
